### PR TITLE
:sparkles: Enable the agent to send event to inventory server

### PR DESCRIPTION
**Pull request by yanmxa**
## Summary

- Remove the ·transport-type· from the parameter/flag
- Determine the ·transport type· by the configuration of the `transport-config` secret
- `ClusterInfo` API - `github.com/stolostron/cluster-lifecycle-api`
- Add inventory client to the generic producer
The full message:
```json
{
   "metadata":{
      "id":0,
      "last_reported":"2024-09-19T04:44:14.89283553Z",
      "resource_type":"k8s-cluster",
      "workspace":"managed-hub",
      "labels":null
   },
   "reporter_data":{
      "reporter_instance_id":"guest",
      "reporter_type":"ACM",
      "last_reported":"2024-09-19T04:44:14.89283553Z",
      "local_resource_id":"1",
      "reporter_version":"0.1",
      "console_href":"https://console-openshift-console.apps.obs-hub-of-hubs-aws-416-sno-69hrc.scale.red-chesterfield.com",
      "api_href":"https://api.obs-hub-of-hubs-aws-416-sno-69hrc.scale.red-chesterfield.com:6443"
   },
   "resource_data":{
      "external_cluster_id":"eeec4c7c-5ede-417c-827e-9a2c583ddef3",
      "cluster_status":"READY",
      "kube_version":"v1.29.5+29c95f3",
      "kube_vendor":"OPENSHIFT",
      "vendor_version":"4.16.0",
      "cloud_platform":"AWS_UPI",
      "nodes":[
         {
            "name":"ip-10-0-4-27.ec2.internal",
            "cpu":"16",
            "memory":"64450660Ki",
            "labels":[
               {
                  "key":"topology.kubernetes.io/zone",
                  "value":"us-east-1a"
               },
               {
                  "key":"failure-domain.beta.kubernetes.io/zone",
                  "value":"us-east-1a"
               },
               {
                  "key":"node.kubernetes.io/instance-type",
                  "value":"m6a.4xlarge"
               },
               {
                  "key":"topology.kubernetes.io/region",
                  "value":"us-east-1"
               },
               {
                  "key":"beta.kubernetes.io/instance-type",
                  "value":"m6a.4xlarge"
               },
               {
                  "key":"failure-domain.beta.kubernetes.io/region",
                  "value":"us-east-1"
               }
            ]
         }
      ]
   }
}
```

## Related issue(s)

Fixes #

## Tests
* [x] Unit/function tests have been added and incorporated into `make unit-tests`.
* [ ] Integration tests have been added and incorporated into `make integration-test`.
* [ ] E2E tests have been added and incorporated into `make e2e-test-all`.
* [x] List other manual tests you have done.
   - Generate the transport-secret with `test/script/event_exporter_inventory.sh`
   - Install the standalone agent(event exporter) with helm chart
   - Consumer the `kessel-inventory` message with `samples/invenotry/consumer/main.go`

### CODE REVIEW

File: pkg/transport/inventory/transfer/k8s_cluster.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+			Metadata: &kessel.Metadata{
+				ResourceType: "k8s-cluster",
+				Workspace:    clusterInfo.Name,
+			},
+			ReporterData: &kessel.ReporterData{
+				ReporterType:       kessel.ReporterData_ACM,
+				ReporterInstanceId: "guest",
+				ReporterVersion:    "0.1",
+				LocalResourceId:    "1",
+				ApiHref:            clusterInfo.Spec.MasterEndpoint,
```
Comment (clyang82) [inline pkg/transport/inventory/transfer/k8s_cluster.go:22]: The local_resource_id is the identification of the resource within the Reporter’s domain. maybe use clusterId here.
ReporterInstanceId should be generated by the system. You can update as CN from client.crt. for example: CN=demo-hub-b-kafka-user

File: pkg/transport/inventory/transfer/k8s_cluster.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+func GetK8SCluster(clusterInfo *clusterinfov1beta1.ManagedClusterInfo) *kessel.CreateK8SClusterRequest {
+	clusterRequest := &kessel.CreateK8SClusterRequest{
+		K8SCluster: &kessel.K8SCluster{
+			Metadata: &kessel.Metadata{
+				ResourceType: "k8s-cluster",
+				Workspace:    clusterInfo.Name,
```
Comment (clyang82) [inline pkg/transport/inventory/transfer/k8s_cluster.go:15]: `Workspace` is useless for ACM.

File: pkg/transport/inventory/client/client.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	if err := evt.DataAs(&data); err != nil {
+		return err
+	}
+
+	for _, clusterInfo := range data {
+		clusterRequest := transfer.GetK8SCluster(&clusterInfo)
+		if clusterRequest != nil {
+			if _, err := client.K8sClusterService.CreateK8SCluster(ctx, clusterRequest); err != nil {
+				return err
+			}
```
Comment (clyang82) [inline pkg/transport/inventory/client/client.go:76]: I am fine for dev-preview to create cloudevents (wrap managedclusterinfo) and then decode events to get managedclusterinfo. Let us observe the inventory-api to response in future.

File: pkg/transport/inventory/transfer/k8s_cluster.go (hunk 2 above)
Comment (yanmxa) [reply pkg/transport/inventory/transfer/k8s_cluster.go]: Done.

File: agent/pkg/status/controller/managedclusters/managed_clsuter_info_syncer_test.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package managedclusters
```
[the remark this answers is not in the document]
Comment (yanmxa) [reply agent/pkg/status/controller/managedclusters/managed_clsuter_info_syncer_test.go]: Done.

[the remark this answers is not in the document]
Comment (yanmxa) [reply agent/pkg/status/controller/managedclusters/managed_clsuter_info_syncer_test.go]: Great! Do you have a plugin that checks the spelling of the file name?

File: pkg/transport/inventory/transfer/k8s_cluster.go (hunk 1 above)
Comment (yanmxa) [reply pkg/transport/inventory/transfer/k8s_cluster.go:21]: As we discussed, I will keep the `LocalReousrceId` with the current value until we have a more explicit definition.

**Comment by yanmxa** [comment]
e2e install the dependencies error, retry again

/test test-e2e

**Pull request merged**